jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java

```diff
@@ -653,15 +653,22 @@ public String toString()
     {
         private final Map<Object, Long> demands = new ConcurrentHashMap<>();
         private final LongConsumer demand = HttpReceiver.this::demand;
-        private final List<Response.DemandedContentListener> listeners;
+        private final List<Response.DemandedContentListener> listeners = new ArrayList<>(2);
 
-        private ContentListeners(List<Response.ResponseListener> responseListeners)
+        private void clear()
         {
-            listeners = new ArrayList<>(responseListeners.size());
-            responseListeners.stream()
-                .filter(Response.DemandedContentListener.class::isInstance)
-                .map(Response.DemandedContentListener.class::cast)
-                .forEach(listeners::add);
+            demands.clear();
+            listeners.clear();
+        }
+
+        private void reset(List<Response.ResponseListener> responseListeners)
+        {
+            clear();
+            for (Response.ResponseListener listener : responseListeners)
+            {
+                if (listener instanceof Response.DemandedContentListener)
+                    listeners.add((Response.DemandedContentListener)listener);
+            }
```

Comment:
re my comment above, this code looks strange to be in a `reset` but reasonable in a `init` method